core/src/test/scala/unit/kafka/common/ZkNodeChangeNotificationListenerTest.scala

```diff
@@ -57,5 +57,12 @@ class ZkNodeChangeNotificationListenerTest extends KafkaServerTestHarness {
 
     zkUtils.createSequentialPersistentPath(seqNodePath, notificationMessage2)
     TestUtils.waitUntilTrue(() => invocationCount == 2 && notification == notificationMessage2, "failed to send/process notification message in the timeout period.")
+
+    (3 to 10).map { i =>
+      zkUtils.createSequentialPersistentPath(seqNodePath, "message" + i)
+      Thread.sleep(50L)
```

Comment:
it was added to make sure, not to purge all previous notifications at the same time. this will create more duplicates in old code. we can remove this sleep and increase expiration period.